crates/subspace-core-primitives/src/objects.rs

```diff
@@ -129,50 +129,55 @@ pub struct PieceObjectMapping {
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Encode, Decode, TypeInfo)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
-pub enum GlobalObject {
-    /// V0 of object mapping data structure
-    #[codec(index = 0)]
-    #[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
-    V0 {
-        /// Piece index where object is contained (at least its beginning, might not fit fully).
-        /// The index and offset must be first, so that the Ord derives are valid.
-        piece_index: PieceIndex,
-        /// Offset of the object in that piece
-        offset: u32,
-        /// Object hash
-        hash: Blake3Hash,
-    },
+pub struct GlobalObject {
+    /// Object hash.
+    /// We order by hash, so object hash lookups can be performed efficiently.
+    hash: Blake3Hash,
+    /// Piece index where object is contained (at least its beginning, might not fit fully)
+    piece_index: PieceIndex,
+    /// Offset of the object in that piece
+    offset: u32,
```

Comment:
This all should be made public properties and all methods removed. The reason we had methods before is to not deal with enum destructuring every time, especially as we add more variants in the future. But now it is taken care of on a different layer, so methods are no longer necessary.